### PR TITLE
User should be able to search other users

### PR DESCRIPTION
**Pull request by azaria-dedmon**
**What does this PR do?**

- This PR provides users the functionality to search other users.

**Description of task to be completed.**

- Provides functionality for users to search other users.

**How should this be manually tested?**

- Run `Python3 run.py`
- Go to `localhost:5000/`
- Click the register link
- After signup, search for users by username in the search bar on the `user home page`

### CODE REVIEW

**Comment by alexkayabula** [comment]
@azaria-dedmon one commit is required for each this pull request.

File: app/__init__.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def logout():
     """Handle user logout."""
     do_logout()
     flash("Success!", "success")
-    return redirect("/")
+    return redirect('/')
```
Comment (alexkayabula) [inline app/__init__.py]: How is this change related to this pull request ?

File: app/__init__.py [hunk 2]
```diff
@@ -109,4 +109,17 @@ def show_state_locations():
     state = request.args.get('state')
 
     locations = get_testing_locations(state, API_BASE_URL)
-    return render_template('location.html', locations=locations)
\ No newline at end of file
+    return render_template('location.html', locations=locations)
```
Comment (alexkayabula) [inline app/__init__.py]: How is this change related to this pull request ?

File: app/templates/users/searched_user.html [hunk 3]
```diff
@@ -2,12 +2,16 @@
...
         <h3>Vaccination Date: {{ searched_user.vax_date }}</h3>
         <h3>Covid Status: {{ searched_user.covid_status }}</h3>
+    {% else %}
+        <h1>No user found</h1>
+    {% endif %}
 </div>
 
-{% endblock %}
\ No newline at end of file
+{% endblock %}
```
Comment (alexkayabula) [inline app/templates/users/searched_user.html]: How is this change related to this pull request.

File: models.py [hunk 4]
```diff
@@ -5,13 +5,13 @@
 
 bcrypt = Bcrypt()  
 db = SQLAlchemy()  
- 
+
```
Comment (alexkayabula) [inline models.py]: How is this change related to this pull request ?

File: models.py [hunk 5]
```diff
@@ -5,13 +5,13 @@
...
 def connect_db(app):    
     db.app = app    
     db.init_app(app)
     db.create_all()
-        
-class User (db.Model):
+
+class User(db.Model):
```
Comment (alexkayabula) [inline models.py]: How is this change related to this pull request ?

File: models.py [hunk 6]
```diff
@@ -42,14 +42,14 @@ class User (db.Model):
         nullable=False  
     )   
     image = db.Column(  
-        db.Text 
+        db.Text
```
Comment (alexkayabula) [inline models.py]: How is this change related to this pull request ?

File: models.py [hunk 7]
```diff
@@ -42,14 +42,14 @@ class User (db.Model):
...
     )   
     state = db.Column(  
         db.Text,    
         nullable=False  
     )   
     vax_date = db.Column(   
-        db.Text 
+        db.Text
```
Comment (alexkayabula) [inline models.py]: How is this change related to this pull request ?

File: models.py [hunk 8]
```diff
@@ -89,7 +89,7 @@ def authenticate(cls, username, password):
         return False
 
 
-class Review (db.Model):   
+class Review(db.Model):
```
Comment (alexkayabula) [inline models.py]: How is this change related to this pull request ?

File: tests/test_user_views.py [hunk 9]
```diff
@@ -157,4 +157,17 @@ def test_map_locations(self):
 
             self.assertEqual(resp.status_code, 200)
 
-            self.assertIn('San Francisco', html)
\ No newline at end of file
+            self.assertIn('San Francisco', html)
```
Comment (alexkayabula) [inline tests/test_user_views.py]: How is this change related to this pull request ?

**Review by alexkayabula — CHANGES_REQUESTED** [review]

**Comment by alexkayabula** [comment]
@azaria-dedmon change the desciption from `Provides functionality for users to search other users in init.py` to `Provides functionality for users to search other users.`. There is no need to specify the file.

**Comment by azaria-dedmon** [comment]
@alexkayabula If I am making minor changes to my code like changing quotation marks from this `' '` to this `" "`, or adding space in my file, am I supposed to be making a whole PR for those adjustments?

**Comment by alexkayabula** [comment]
> @alexkayabula If I am making minor changes to my code like changing quotation marks from this `' '` to this `" "`, or adding space in my file, am I supposed to be making a whole PR for those adjustments?

Yeah...any change to the code base not related to the PR you are working on (Regardless whether it is a space, comma etc) should be in a different PR.

**Comment by alexkayabula** [comment]
@azaria-dedmon there should be one commit message per PR. Kindly make the necessary change.

File: tests/test_user_views.py (hunk 9 above)
Comment (alexkayabula) [inline tests/test_user_views.py]: How is this change related to the PR ?

**Review by alexkayabula — CHANGES_REQUESTED** [review]

File: app/__init__.py [hunk 10]
```diff
@@ -109,4 +109,14 @@ def show_state_locations():
...
```
Comment (alexkayabula) [inline app/__init__.py]: How is this change related to this PR ?

File: tests/test_user_views.py (hunk 9 above)
Comment (alexkayabula) [inline tests/test_user_views.py]: How is this change related to this PR ?

**Review by alexkayabula — CHANGES_REQUESTED** [review]

File: app/__init__.py (hunk 10 above)
Comment (azaria-dedmon) [reply app/__init__.py]: I'm having trouble understanding the issue with this code. I am adding new code underneath `return render_template('location.html', locations=locations)`. Maybe that is why this line of code keeps showing up? I am not adding anything new to the mentioned line.

File: tests/test_user_views.py (hunk 9 above)
Comment (azaria-dedmon) [reply tests/test_user_views.py]: The same applies here. I am not adding anything new/modifying ` self.assertIn('San Francisco', html)` i'm only adding code beneath this line.

File: app/__init__.py (hunk 10 above)
Comment (alexkayabula) [reply app/__init__.py]: I guess this issue is still under review [https://github.com/rtfpessoa/diff2html/issues/157](https://github.com/rtfpessoa/diff2html/issues/157) We shall get back to it later.

File: tests/test_user_views.py (hunk 9 above)
Comment (alexkayabula) [reply tests/test_user_views.py]: I guess this issue is still under review [https://github.com/rtfpessoa/diff2html/issues/157](https://github.com/rtfpessoa/diff2html/issues/157) We shall get back to it later.

Comment (alexkayabula) [reply tests/test_user_views.py]: I guess this issue is still under review [https://github.com/rtfpessoa/diff2html/issues/157](https://github.com/rtfpessoa/diff2html/issues/157) We shall get back to it later.

**Comment by alexkayabula** [comment]
LGTM.
